flink-core/src/main/java/org/apache/flink/util/NetUtils.java

```diff
@@ -104,7 +104,10 @@ public static InetSocketAddress parseHostPortAddress(String hostPort) {
      */
     private static URL validateHostPortString(String hostPort) {
         try {
-            URL u = new URL("http://" + hostPort);
+            URL u =
+                    (hostPort.startsWith("http://") || hostPort.startsWith("https://"))
```

Comment:
Some places we use `equalsIgnoreCase` for protocol but here not allowed. It would be good to make it consistent. Either not allow or allow all the places.